source/Ship.cpp

```diff
@@ -275,29 +275,60 @@ void Ship::Load(const DataNode &node)
 				if(child.Size() >= 2)
 					outfit = GameData::Outfits().Get(child.Token(1));
 			}
-			Angle gunPortAngle = Angle(0.);
-			bool gunPortParallel = false;
+			Hardpoint::BaseAttributes attributes;
+			attributes.baseAngle = Angle(0.);
+			attributes.isParallel = false;
+			attributes.isOmnidirectional = true;
 			bool drawUnder = (key == "gun");
 			if(child.HasChildren())
 			{
+				bool defaultBaseAngle = true;
 				for(const DataNode &grand : child)
 				{
+					bool needToCheckAngles = false;
 					if(grand.Token(0) == "angle" && grand.Size() >= 2)
-						gunPortAngle = grand.Value(1);
+					{
+						attributes.baseAngle = grand.Value(1);
+						needToCheckAngles = true;
+						defaultBaseAngle = false;
+					}
 					else if(grand.Token(0) == "parallel")
-						gunPortParallel = true;
+						attributes.isParallel = true;
+					else if(grand.Token(0) == "arc" && grand.Size() >= 3)
+					{
+						attributes.isOmnidirectional = false;
+						attributes.arc = make_pair(Angle(grand.Value(1)), Angle(grand.Value(2)));
+						needToCheckAngles = true;
+					}
 					else if(grand.Token(0) == "under")
 						drawUnder = true;
 					else if(grand.Token(0) == "over")
 						drawUnder = false;
 					else
-						grand.PrintTrace("Skipping unrecognized attribute:");
+						grand.PrintTrace("Warning: Child nodes of \"" + key
+							+ "\" tokens can only be \"angle\", \"parallel\", or \"arc\":");
+
+					if(needToCheckAngles && !defaultBaseAngle && !attributes.isOmnidirectional)
+					{
+						attributes.arc.first += attributes.baseAngle;
+						attributes.arc.second += attributes.baseAngle;
```

Comment:
This means that the `arc` given in is relative to the angle specified, right?

Shouldn't the angle instead just have the middle range between the two elements added to it, instead?

Reply:
If I understand it right that's how the previous PR worked and it doesn't seems to work for turrets pointing backwards. It's also more intuitive to have the a base angle then how much it can traverse to the left and how much it can traverse to the right. It also allows turrets to bias towards on way or the other by default instead of being forced to aim at the center of the arc when idle.

Reply:
I just wanted to be sure that part was clear, you can resolve this.